frontend/src/state/backend.ts

```diff
@@ -1,36 +1,71 @@
-import { all, call, takeEvery } from 'redux-saga/effects'
+import { all, call, takeEvery, put, delay } from 'redux-saga/effects'
+import { updateQueueCreator } from './queue';
 
 const CREATE_QUEUE = "@backend/CREATE_QUEUE";
-
-export const createQueueCreator = (placeId: string, name: string) => ({
+interface CreateQueueAction {
+  type: "@backend/CREATE_QUEUE";
+  placeId: string;
+  name: string;
+}
+export const createQueueCreator = (placeId: string, name: string): CreateQueueAction => ({
   type: CREATE_QUEUE,
   placeId,
   name
 });
 
-function* queueSaga(action: any) {
+const FETCH_QUEUES = "@backend/FETCH_QUEUES";
+interface FetchQueuesAction {
+  type: "@backend/FETCH_QUEUES";
+  placeId: string;
+}
+export const fetchQueuesCreator = (placeId: string): FetchQueuesAction => ({
+  type: FETCH_QUEUES,
+  placeId
+});
+
+type backendAction = CreateQueueAction | FetchQueuesAction;
+
+function* queueSaga(action: backendAction) {
   if (action.type === CREATE_QUEUE) {
-    try {
-      const response = yield call(fetch, `/places/${action.placeId}/queues`, {
-        method: "POST",
-        headers: {
-          'Content-Type': 'application/json'
-        },
-        body: JSON.stringify({queueName: action.name})
-      });
-      if (response.ok) {
-        console.log("Queue created, response: ", response);
-      } else {
-        console.error("Queue creation failed: ", response);
+    const response = yield call(fetch, `/places/${action.placeId}/queues`, {
+      method: "POST",
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ queueName: action.name })
+    });
+    if (response.ok) {
+      console.log("Queue created, response: ", response);
+    } else {
+      console.error("Queue creation failed: ", response);
+    }
+  } else if (action.type === FETCH_QUEUES) {
+    const response = yield call(fetch, `/places/${action.placeId}/queues`, {
```

Comment:
Without having any idea of TypeScript, shouldn't the query option of .../queues?personDetails=short or full be somewhere here?

Reply:
Yeah, let's fix that tomorrow.

Reply:
full is not yet implemented in the backend. will use short for now.